tslib/src/functions/check-notifier/check-notifier.v1.ts

```diff
@@ -0,0 +1,31 @@
+import { ICheckNotificationMessage, CheckNotificationType } from './check-notification-message'
+import { CheckNotifierDataService, ICheckNotifierDataService } from './check-notifier.data.service'
+
+export class CheckNotifier {
+  private checkNotifierDataService: ICheckNotifierDataService
+
+  constructor (checkNotifierDataService?: ICheckNotifierDataService) {
+    if (checkNotifierDataService === undefined) {
+      checkNotifierDataService = new CheckNotifierDataService()
+    }
+    this.checkNotifierDataService = checkNotifierDataService
+  }
+
+  async notify (notification: ICheckNotificationMessage) {
+    switch (notification.notificationType) {
+      case CheckNotificationType.checkInvalid:
+        throw new Error('not yet implemented')
+      case CheckNotificationType.checkComplete:
+        return this.checkNotifierDataService.updateCheckAsComplete(notification.checkCode)
+    }
+  }
+}
+
+export enum CheckStatus {
+  New = 1,
+  // Expired = 2, To be removed
```

Comment:
What happens to NEW  checks that are not collected?

Reply:
they would remain NEW, never collected.  Does anything else actually need to be done?
we can discuss this in more detail later today if you wish.

Reply:
what's the outcome of that comment?

Reply:
it's outside of this PR, but we will be addressing it soon.  It was discussed on this mornings planning session and will emerge with the pupil state analysis.